numba/np/arraymath.py

```diff
@@ -831,8 +831,8 @@ def iscomplexobj(x):
     # Implementation based on NumPy
     # https://github.com/numpy/numpy/blob/d9b1e32cb8ef90d6b4a47853241db2a28146a57d/numpy/lib/type_check.py#L282-L320
     if isinstance(x, types.Optional):
-        msg = 'Expected argument {} but got Optional[{}, None]'
-        raise TypingError(msg.format(x.key, x.key))
+        msg = f'Expected argument {x.key} but got {x}'
+        raise TypingError(msg)
```

Comment:
Is this branch tested?

Reply:
Hum....no. Is there a way to create a variable with type `Optional`?

Reply:
try something like `y = x if x else None` as long as `x` is not known at compile time.

Reply:
I've changed the code to accommodate optional types. If CI passes, the PR will be ready for another review